dlt/destinations/impl/motherduck/configuration.py

```diff
@@ -32,11 +35,38 @@ def _token_to_password(self) -> None:
         if self.query and "token" in self.query:
             self.password = TSecretValue(self.query.pop("token"))
 
+    def _connect(self, read_only: bool) -> Any:
```

Comment:
This is mostly copy-paste from the base client of our `duckdb` destination because the whole logic was hard to compose I decided to make it separate for motherduck